### PR TITLE
feat: getProducts api-client

### PR DESCRIPTION
**Pull request by zoltantakacs-born**
getProducts function to request products from bigcommerce API.

## Related Issue
[VBC-21](https://borngroup.atlassian.net/browse/VBC-21)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/api-client/src/types/api/products.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    | 'is_visible'
+    | 'total_sold';
+  sku?: string;
+  'sku:in'?: string[];
+};
+
+/**
+ * Definition of product images
+ */
+type Image = {
```
Comment (WojtekTheWebDev) [inline packages/api-client/src/types/api/products.ts:69]: Can you extract this part and import it from another file like api-client/src/types/api/product/image.ts ?

**Review by WojtekTheWebDev — APPROVED** [review]
Only 1 small remark in the comment

File: packages/composables/src/getters/productGetters.ts [hunk 2]
```diff
@@ -42,25 +42,7 @@ function getCoverImage(product: Product): string {
...
-      images: [
-        'https://s3-eu-west-1.amazonaws.com/commercetools-maximilian/products/081223_1_large.jpg'
-      ],
-      price: {
-        original: 12.34,
-        current: 10.00
-      }
-    }
-  ];
+  return products ?? [];
```
Comment (scernisevs) [inline packages/composables/src/getters/productGetters.ts:45]: hi @zoltantakacs-born ,

Could we please add a "TODO" comment in here, to come back to this function together with Configurable Products and product variation development

**Review by WojtekTheWebDev — APPROVED** [review]

File: packages/api-client/src/api/product/get/index.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  params: GetProductsProps
+): Promise<ProductsResponse> {
+  if (params.include && !params.include.includes('images')) {
+    params.include = params.include.concat(',images');
+  } else if (!params.include) {
+    params.include = 'images';
+  }
+
+  return await context.client.get(
+    queryString.stringifyUrl({ url: '/catalog/products', query: params })
```
Comment (dhargitai) [inline packages/api-client/src/api/product/get/index.ts:19]: Hey Zoli, I've been asked to use a separate file to store the string constants for the API endpoints.
Maybe could you use it as well?

https://github.com/vuestorefront/bigcommerce/pull/17/files#diff-e18d8be656d639eaa5fc9cf089de6976b1300430a06b81bdb2d98345d5e2a03d

File: packages/api-client/src/types/api/endpoints/index.ts [hunk 4]
```diff
@@ -1,4 +1,18 @@
...
 /**
  * Definition of all API-client methods available in context.
  */
-export type Endpoints = Record<string, string>;
+export type Endpoints = {
+
+  /**
+   * Returns a list of Products.
+   * @param {BigcommerceIntegrationContext} context An auto-generated value prepended to the method as a first parameter.
+   * @param {GetProductsProps} props An object containinig the filters for products.
```
Comment (dhargitai) [inline packages/api-client/src/types/api/endpoints/index.ts:12]: There's a typo here.

**Review by dhargitai — CHANGES_REQUESTED** [review]

**Review by scernisevs — APPROVED** [review]
LGTM! :) thank you.

**Review by WojtekTheWebDev — APPROVED** [review]
LGTM, in my other PR I also created types for meta and pagination so we would need to solve the conflicts after merging but nice pr ;)

**Review by scernisevs — CHANGES_REQUESTED** [review]
LGTM! Thank you :) 
Only the 2 outstanding comments raised by @dhargitai are still needed before this can be merged in!

**Review by dhargitai — APPROVED** [review]
Looks good! :)

**Review by scernisevs — APPROVED** [review]

**Pull request merged**